x/dex/keeper/msg_server.go

```diff
@@ -215,6 +217,9 @@ func (k MsgServer) MultiHopSwap(
 	goCtx context.Context,
 	msg *types.MsgMultiHopSwap,
 ) (*types.MsgMultiHopSwapResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
```

Comment:
Sorry, I know the naming is a bit confusing. But we want to track gas anytime we are doing a swap (LimitOrders + multihop swap_. The correct place to do that is here https://github.com/neutron-org/neutron/blob/9d2df9d1ba8d34591b89c1eaeecc426d4acd4f97/x/dex/keeper/liquidity.go#L11